autoload/fern/internal/drawer/hover_popup.vim

```diff
@@ -1,63 +1,81 @@
+  if !s:available()
+    call fern#logger#warn('hover popup is not supported, popup_create() or nvim environment
           \ does not exist. Disable this message
           \ with g:fern#disable_drawer_hover_popup.')
     return
   endif
 
   augroup fern_internal_drawer_hover_popup_init
     autocmd! * <buffer>
-    autocmd CursorMoved <buffer> call s:cursor_moved_event()
+    autocmd CursorMoved <buffer> call timer_stop(s:timer) | call s:hide() | let s:timer = timer_start(g:fern#drawer_hover_popup_delay, { -> s:show(fern#helper#new()) })
+    autocmd BufLeave <buffer> call timer_stop(s:timer) | call s:hide()
   augroup END
 endfunction
 
-function! fern#internal#drawer#hover_popup#calculate_node_char_offset(node) abort
-  " find line offset where label text begins
-  let line = getline('.')
-  let labelbegin = charidx(line, strridx(line, a:node.label))
-  let labelbegin = labelbegin < 0 ? 0 : labelbegin
-
-  let windowid = win_getid()
-
-  " get cursor position in drawer window (char- and byte-indexed)
-  let charpos = getcursorcharpos(windowid)
-  let pos = getcurpos(windowid)
-
-  " get cursor position relative to screen
-  let cursorpos = screenpos(windowid, pos[1], pos[2])
-
-  " calculate screen column where label text begins
-  return cursorpos['col'] - charpos[2] + labelbegin
-endfunction
-
-function! fern#internal#drawer#hover_popup#should_display_popup() abort
-  return len(getline('.')) >= winwidth(0)
+function! s:available() abort
+  return has('nvim') || exists('*popup_create')
 endfunction
 
-function! s:cursor_moved_event() abort
-  let helper = fern#helper#new()
-
-  if fern#internal#drawer#hover_popup#should_display_popup()
-    call s:show_popup(helper)
+function! s:show(helper) abort
+  if strdisplaywidth(getline('.')) <= winwidth(0)
+    return
   endif
-endfunction
 
-function! s:show_popup(helper) abort
   let node = a:helper.sync.get_cursor_node()
   if node is# v:null
     return
   endif
 
-  let label_offset = fern#internal#drawer#hover_popup#calculate_node_char_offset(node)
-  call popup_create(l:node.label, {
-        \ 'line': 'cursor',
-        \ 'col': label_offset + 1,
-        \ 'moved': 'any',
-        \})
+  let line = getline('.')
+  if has('nvim')
+    let s:win = nvim_open_win(nvim_create_buf(v:false, v:true), v:false, {
+    \    'relative': 'win',
+    \    'bufpos': [line('.') - 2, 0],
+    \    'width': strdisplaywidth(substitute(line, '[^[:print:]]*$', '', 'g')),
+    \    'height': 1,
+    \    'noautocmd': v:true,
+    \    'style': 'minimal',
+    \  })
+  else
+    let ui_width = screenpos(0, line('.'), 1).col - win_screenpos(0)[1]
+    let s:win = popup_create(line, {
+    \    'line': 'cursor',
+    \    'col': ui_width + 1,
+    \    'maxwidth': strdisplaywidth(substitute(line, '[^[:print:]]*$', '', 'g')),
+    \  })
+  endif
+  function! s:apply() abort closure
+    call setbufline('%', 1, line)
+    setlocal nowrap
+    setlocal cursorline
```

Comment:
I don't know popup/hover window well so I'm not sure if it's needed but don't we need to set options like?

```vim
setlocal noswap nobuflisted nolist
setlocal buftype=nofile
```